### PR TITLE
Rephrase paragraph

### LINKED ISSUE #69: Clarification of Chapter 3 Memory and Allocation
I am not fully understanding this paragraph:

https://github.com/pedropark99/zig-book/blob/3a02942229000e0bea213f4fdd6d79ba4f57b3c6/Chapters/01-memory.qmd#L26-L28

Do you mean every program needs memory? You switch between computers and programs e.g. in the last sentence you are talking about programming languages. Maybe rephrasing it something like this would make it clearer:

> [!NOTE]
Computers fundamentally rely on memory to function. This memory acts as a temporary storage space for the data and results generated during computations.  Without memory, the core concepts of programming languages, such as "variables" and "objects" which hold these values, would be impossible. In essence, memory provides the foundation for data manipulation and storage within a computer system, enabling the execution of programs written in any language.

### PR DESCRIPTION
**Pull request by pedropark99**
This PR seeks to rephrase a paragraph to fix #69 reported by @kurz-m .

### CODE REVIEW

**Comment by pedropark99** [comment]
@kurz-m  Hey! You can review this PR (if you want to) and suggest changes in case the rephrasing can still be improved.

Don't feel pressured to review this PR, I just thought you might like to review it.

**Comment by kurz-m** [comment]
@pedropark99 thanks ;) I would be happy to review it.

File: Chapters/01-memory.qmd [hunk 1]
```diff
@@ -23,9 +23,11 @@ In this chapter, we will talk about memory. How does Zig controls memory? What
...
 different/special in Zig? You will find the answers here.
 
-Every computer needs memory. Is by having memory that computers can temporarily store
-the values/results of your calculations. Without memory, programming languages would never have
-concepts such as "variables", or "objects", to store the values that you generate.
+Computers fundamentally rely on memory to function. Is by having memory that computers
+can temporarily store the data and values generated during computations
+the values/results of your calculations. Without memory, the core concepts of
+"variables" and "objects" in programming languages would be impossible.
+
```
Comment (kurz-m) [inline Chapters/01-memory.qmd:30]: I would change it to the following:

Computers fundamentally rely on memory to function.
This memory acts as a temporary storage space for the data and values generated during computations. 
Without memory, the core concepts of "variables" and "objects" in programming languages would be impossible.

**Review by kurz-m — COMMENTED** [review]
I added my thoughts on how I would change the sentence for better flow and clarity of this sentence. Thanks for updating this section! It's much appreciated.

File: Chapters/01-memory.qmd (hunk 1 above)
Comment (pedropark99) [reply Chapters/01-memory.qmd]: Done!

**Review by kurz-m — APPROVED** [review]
I think those changes make the intention of this section more clear. Thanks for the updates!

**Review by kurz-m — APPROVED** [review]

**Pull request merged**